### PR TITLE
Update script example in logging FAQ

### PR DESCRIPTION
**Pull request by kingthorin**
Here is the test output based on running the spider, turning on DEBUG then running again, returning to INFO and running again:
<details>
<summary>Example output:</summary>

```console
109554 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Starting spidering scan on http://localhost/test.html at 2022-01-28T11:08:28.947-0500
109558 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.spider.Spider - Spider initializing...
109565 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.spider.Spider - Starting spider...
109662 [ZAP-SpiderThreadPool-0-thread-2] INFO  org.zaproxy.zap.spider.Spider - Spidering process is complete. Shutting down...
109664 [ZAP-SpiderShutdownThread-0] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Spider scanning complete: true on http://localhost/test.html at 2022-01-28T11:08:29.056-0500
120698 [ZAP-SpiderInitThread-1] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Starting spidering scan on http://localhost/test.html at 2022-01-28T11:08:40.091-0500
120699 [ZAP-SpiderInitThread-1] INFO  org.zaproxy.zap.spider.Spider - Spider initializing...
120700 [ZAP-SpiderInitThread-1] INFO  org.zaproxy.zap.spider.Spider - Starting spider...
120718 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link01.svg
120719 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link01.svg
120720 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://www.w3.org from: http://localhost/link01.svg
120720 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using 'http://www.w3.org': http://www.w3.org/
120724 [ZAP-SpiderThreadPool-1-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/lighttpd.svg
120735 [ZAP-SpiderThreadPool-1-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link03.svg
120735 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link02.svg
120736 [ZAP-SpiderThreadPool-1-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link03.svg
120736 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link02.svg
120736 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://localhost/i.html from: http://localhost/link02.svg
120736 [ZAP-SpiderThreadPool-1-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://localhost/alert.html from: http://localhost/link03.svg
120736 [ZAP-SpiderThreadPool-1-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using 'i.html': http://localhost/i.html
120736 [ZAP-SpiderThreadPool-1-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using '/alert.html': http://localhost/alert.html
120767 [ZAP-SpiderThreadPool-1-thread-1] INFO  org.zaproxy.zap.spider.Spider - Spidering process is complete. Shutting down...
120768 [ZAP-SpiderShutdownThread-1] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Spider scanning complete: true on http://localhost/test.html at 2022-01-28T11:08:40.161-0500
320292 [ZAP-SpiderInitThread-2] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Starting spidering scan on http://localhost/test.html at 2022-01-28T11:11:59.685-0500
320292 [ZAP-SpiderInitThread-2] INFO  org.zaproxy.zap.spider.Spider - Spider initializing...
320293 [ZAP-SpiderInitThread-2] INFO  org.zaproxy.zap.spider.Spider - Starting spider...
320360 [ZAP-SpiderThreadPool-2-thread-1] INFO  org.zaproxy.zap.spider.Spider - Spidering process is complete. Shutting down...
320361 [ZAP-SpiderShutdownThread-2] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Spider scanning complete: true on http://localhost/test.html at 2022-01-28T11:11:59.754-0500
```

</details>

### CODE REVIEW

**Comment by kingthorin** [comment]
Submitted via the web UI, set DCO pass. I do agree with https://developercertificate.org/ and have signed off on all recent PRs (submitted remotely).

**Comment by thc202** [comment]
Getting the same behaviour with existing script.

**Comment by kingthorin** [comment]
Wait the same as in everything is set to debug or the same as the example above?

**Comment by thc202** [comment]
I tried with the spider package, INFO and DEBUG, and it applied just to the spider classes. I did not try just with one class, let me try.

**Comment by thc202** [comment]
Yeah, what you were saying it changes other classes as well.

**Comment by kingthorin** [comment]
I think it might have to do with the existing examples. Maybe one of the paths is no longer valid so it’s applying it to a parent path/location.

I’ll look into that further.

**Comment by kingthorin** [comment]
Okay cross that off the list. Those packages are still the same, the setting is applied too broadly.

File: site/content/faq/how-do-you-configure-zap-logging.md [hunk 1]
```diff
@@ -39,16 +39,14 @@ Logging can be selectively enabled using a [Stand Alone script](/docs/desktop/ad
...
 
 // The following will enable DEBUG logging for the API
-config.getLoggerConfig("org.zaproxy.zap.extension.api.API").setLevel(Level.DEBUG)
+Configurator.setLevel("org.zaproxy.zap.extension.api.API", Level.DEBUG);
 // The following will enable DEBUG logging for the Session Fixation scan rule
-config.getLoggerConfig("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule").setLevel(Level.DEBUG)
-
-context.updateLoggers()
+Configurator.setLevel("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule", Level.DEBUG);
+// The following well enable DEBUG logging for the spider add-on
```
Comment (kingthorin) [inline site/content/faq/how-do-you-configure-zap-logging.md:50]: Just noticed the typo here, will correct in a bit

**Comment by kingthorin** [comment]
When setting for an individual class:

```console
113543 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Starting spidering scan on http://localhost/test.html at 2022-01-28T15:48:20.048-0500
113548 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.spider.Spider - Spider initializing...
113558 [ZAP-SpiderInitThread-0] INFO  org.zaproxy.zap.spider.Spider - Starting spider...
113595 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link01.svg
113602 [ZAP-SpiderThreadPool-0-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/lighttpd.svg
113618 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link01.svg
113618 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://www.w3.org from: http://localhost/link01.svg
113618 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using 'http://www.w3.org': http://www.w3.org/
113632 [ZAP-SpiderThreadPool-0-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link02.svg
113633 [ZAP-SpiderThreadPool-0-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link02.svg
113633 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - SVG Spider attempting to parse http://localhost/link03.svg
113634 [ZAP-SpiderThreadPool-0-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://localhost/i.html from: http://localhost/link02.svg
113634 [ZAP-SpiderThreadPool-0-thread-2] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using 'i.html': http://localhost/i.html
113635 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Identified 1 nodes with href attribute from: http://localhost/link03.svg
113635 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Resolved URL: http://localhost/alert.html from: http://localhost/link03.svg
113635 [ZAP-SpiderThreadPool-0-thread-1] DEBUG org.zaproxy.addon.spider.SvgHrefSpider - Canonical URL constructed using '/alert.html': http://localhost/alert.html
113682 [ZAP-SpiderThreadPool-0-thread-1] INFO  org.zaproxy.zap.spider.Spider - Spidering process is complete. Shutting down...
113684 [ZAP-SpiderShutdownThread-0] INFO  org.zaproxy.zap.extension.spider.SpiderThread - Spider scanning complete: true on http://localhost/test.html at 2022-01-28T15:48:20.189-0500
150952 [AWT-EventQueue-0] INFO  org.parosproxy.paros.control.Control - Discard Session
151395 [ZAP-Shutdown] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit start
151401 [ZAP-Shutdown] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - dataFileCache commit end
151406 [ZAP-Shutdown] INFO  hsqldb.db.HSQLDB379AF3DEBD.ENGINE - Database closed
151531 [ZAP-Shutdown] INFO  org.parosproxy.paros.control.Control - OWASP ZAP Dev Build terminated.
```

File: site/content/faq/how-do-you-configure-zap-logging.md [hunk 2]
```diff
@@ -39,16 +39,14 @@ Logging can be selectively enabled using a [Stand Alone script](/docs/desktop/ad
 running (the example below is a JavaScript script):
 
 ```JavaScript
-var Level = Java.type("org.apache.logging.log4j.Level")
-var LoggerContext = Java.type("org.apache.logging.log4j.core.LoggerContext")
-
-var context = LoggerContext.getContext()
-var config = context.getConfiguration()
+// https://logging.apache.org/log4j/log4j-2.4/faq.html#reconfig_level_from_code
```
Comment (thc202) [inline site/content/faq/how-do-you-configure-zap-logging.md:42]: Maybe link to the JavaDocs? (To not link to old version.)

File: site/content/faq/how-do-you-configure-zap-logging.md [hunk 3]
```diff
@@ -39,16 +39,14 @@ Logging can be selectively enabled using a [Stand Alone script](/docs/desktop/ad
...
 // The following will enable DEBUG logging for the API
-config.getLoggerConfig("org.zaproxy.zap.extension.api.API").setLevel(Level.DEBUG)
+Configurator.setLevel("org.zaproxy.zap.extension.api.API", Level.DEBUG);
 // The following will enable DEBUG logging for the Session Fixation scan rule
-config.getLoggerConfig("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule").setLevel(Level.DEBUG)
-
-context.updateLoggers()
+Configurator.setLevel("org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule", Level.DEBUG);
+// The following will enable DEBUG logging for the spider add-on
+Configurator.setLevel("org.zaproxy.addon.spider", Level.DEBUG);
```
Comment (thc202) [inline site/content/faq/how-do-you-configure-zap-logging.md:51]: I'd suggest doing it in one go, e.g.:
```JavaScript
Configurator.setLevel({
  // The following will enable DEBUG logging for the API
  "org.zaproxy.zap.extension.api.API" : Level.DEBUG,
  // The following will enable DEBUG logging for the Session Fixation scan rule
  "org.zaproxy.zap.extension.ascanrulesBeta.SessionFixationScanRule" : Level.DEBUG,
  // The following will enable DEBUG logging for the spider add-on
  "org.zaproxy.addon.spider" : Level.DEBUG,
});
```
Less verbose and more efficient as it only updates the loggers once.

**Comment by thc202** [comment]
The corresponding Stand Alone script in the Community Scripts repo needs to be fixed as well.

**Comment by kingthorin** [comment]
Will tweak that and open a second PR.

**Comment by kingthorin** [comment]
Updated.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by thc202** [comment]
The imports were removed accidentally (also missing in the other PR).

**Comment by kingthorin** [comment]
Fixed

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you again :)

**Review by psiinon — APPROVED** [review]

**Pull request merged**